streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -1103,6 +1133,7 @@ private void describeSubtopology(final TopologyDescription description,
                 final AbstractNode predecessor = nodesByName.get(predecessorName);
                 node.addPredecessor(predecessor);
                 predecessor.addSuccessor(node);
+                updateSize(predecessor, node.size);
```

Comment:
I am not sure, if this will do the update correctly. There is no ordering guarantee for the outer loop; IMHO, this could lead to double counting?

Maybe it's not an issue though, as the counts are never exposed to the users and only leveraged internally to sort nodes (ie, as long as the numbers are larger/smaller in relation to each other, the final sort is the same). But I am not sure.

Reply:
I thought about it when working on the PR, and I think it does not matter: as long as the `updateSize` is called once and only once on each node, the resulted count should be correct, as for each node, its count is always incremented by one for each of its sub-tree's successor nodes.

Reply:
It does not always add "1" -- it adds `node.size` -- but maybe a fix would be to remove `delta` parameter and only add `1` -- as `updateSize` is recursive this sound correct. Right now, you would double count:

`A(1) -> B(1) -> C(1)` all initialized with `1`. First update `C` adds `1` to `B` and `A` resulting in `A(2) -> B(2) -> C(1)`. Next, `B` is updated, adding `2 to `A` resulting in `A(4) -> B(2) -> C(1)` what is not correct (but maybe sufficient for ordering anyway -- but not sure.)

Reply:
That is actually not true :) If you connect B as C's predecessor first, then it would be 

```
A(1)  B(2) -> C(1)
```

Since A is not added as B's predecessor yet, and then later when you connect A as B's predecessor, it would then be

```
A(3) -> B(2) -> C(1)
```

Reply:
Ack. Got it :)